Servers/UI/OJS.Servers.Ui/ClientApp/src/components/home-contests/ContestsList.tsx

```diff
@@ -22,12 +24,32 @@ const ContestsList = ({
     contests,
     contestStatus,
 }: IContestsListProps) => {
-    const renderContest = useCallback(
-        (contest: IIndexContestsType) => (
-            <ContestCard contest={contest} />
-        ),
-        [],
-    );
+    const [ contestItemsToShow, setContestItemsToShow ] = React.useState(4);
+
+    const handleResize = () => {
+        if (window.innerWidth <= 950) {
+            setContestItemsToShow(1);
+        } else if (window.innerWidth <= 1350) {
+            setContestItemsToShow(2);
+        } else if (window.innerWidth <= 1540) {
+            setContestItemsToShow(3);
+        } else {
+            setContestItemsToShow(4);
+        }
+    };
+
+    React.useEffect(() => {
+        handleResize();
+    });
+
+    React.useEffect(() => {
+        handleResize();
```

Comment:
Isn't this duplicate call of handleResize, as you call it in the above `useEffect`

Reply:
it's not, this effect is called only when the component is rendered, which initially load 4 items on the row by default, and if you open it in smaller resolution it will load 4 items, not taking into consideration the width of the screen, that's where this effect comes into action, having to set the appropriate number of displayed items based on the width of the screen of the device the user is opening the page on initially.